qiskit/transpiler/passes/mapping/full_ancilla_allocation.py

```diff
@@ -77,10 +77,11 @@ def run(self, dag):
 
         if idle_physical_qubits:
             if self.ancilla_name in dag.qregs:
-                save_prefix = QuantumRegister.prefix
+                previous_prefix = QuantumRegister.prefix
                 QuantumRegister.prefix = self.ancilla_name
                 qreg = QuantumRegister(len(idle_physical_qubits))
-                QuantumRegister.prefix = save_prefix
+                _ = qreg.name
```

Comment:
this is an example of the weird behavior i pointed to above. it is very non-transparent that this statement sets a name.

Reply:
I agree that's non-transparent, can removed lazy name generation if that's the argument. It will come with a small performance penalty.

Reply:
removed in f95d02b

Reply:
Is the line `_ = qreg.name` still needed?